File: Monotonic_Stack_29_Oct_2023/0739_daily_temperatures.py
class Solution:
    def dailyTemperatures(self, t: List[int]) -> List[int]:
        s = []
        ans = [0] * len(t)
        
        for i in range(len(t)):
            while s and t[s[-1]] < t[i]:
                idx = s.pop()
                ans[idx] = i - idx
            s.append(i)

        return ans

File: Monotonic_Stack_29_Oct_2023/0739-daily-temperatures.go
package main

func dailyTemperatures(T []int) []int {
	res := make([]int, len(T))
	s := make([]int, 0, len(T))

	for i, t := range T {
		for top := len(s)-1; top >= 0 && T[s[top]] < t; top-- {
			res[s[top]] = i - s[top]
			s = s[:top]
		}
		s = append(s, i)
	}

	return res
}
